### PR TITLE
Add a CRD field to replace InsecureSkipVerify=true

### PR DESCRIPTION
**Pull request by bdunne**
~~Test image: docker.io/bdunne/manageiq-operator:ssl_verify~~
Add optional CR value `OIDCOAuthIntrospectionSSLVerify` to determine whether or not to verify SSL when fetching the introspection URL.  Default to SSLVerify for safety.

### CODE REVIEW

**Review by kbrock — APPROVED** [review]
LGTM

Are you thinking of passing a boolean parameter for passing in `InsecureSkipVerify:true` but defaulting to `false`?
Or are you thinking we can just drop it?

**Comment by Fryguy** [comment]
I had the same q as @kbrock.  Do we need a setting for users that might need to set verify false?

File: manageiq-operator/api/v1alpha1/helpers/miq-components/httpd.go [hunk 1]
```diff
@@ -659,9 +659,9 @@ func tlsSecretName(cr *miqv1alpha1.ManageIQ) string {
 	return secretName
 }
 
-func fetchIntrospectionUrl(providerUrl string) (string, error) {
+func fetchIntrospectionUrl(providerUrl string, sslVerify bool) (string, error) {
 	customTransport := http.DefaultTransport.(*http.Transport).Clone()
-	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: !sslVerify}
```
Comment (Fryguy) [inline manageiq-operator/api/v1alpha1/helpers/miq-components/httpd.go:664]: This is fine, but can we also just not set it?  Something like

```suggestion
	if !sslVerify {
		customTransport.TLSClientConfig = &tls.Config{InsecureSkipVerify: !sslVerify}
	}
```

Comment (Fryguy) [reply manageiq-operator/api/v1alpha1/helpers/miq-components/httpd.go:664]: Merging anyway.

**Comment by Fryguy** [comment]
This PR has been effectively backported to `radjabov` via the merge of master into radjabov.

**Comment by bdunne** [comment]
Unclean backport... Trying in https://github.com/ManageIQ/manageiq-pods/pull/1128 but tests aren't passing yet.

**Comment by Fryguy** [comment]
@bdunne I moved this to morphy/yes? so it wouldn't get in the way of other backports.

**Pull request merged**